### PR TITLE
[SPARK-23633][SQL] Update Pandas UDFs section in sql-programming-guide

### PR DESCRIPTION
**Pull request by icexelloss**
## What changes were proposed in this pull request?

Update Pandas UDFs section in sql-programming-guide. Add section for grouped aggregate pandas UDF.

### CODE REVIEW

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -1804,6 +1804,25 @@ The following example shows how to use `groupby().apply()` to subtract the mean
 For detailed usage, please see [`pyspark.sql.functions.pandas_udf`](api/python/pyspark.sql.html#pyspark.sql.functions.pandas_udf) and
 [`pyspark.sql.GroupedData.apply`](api/python/pyspark.sql.html#pyspark.sql.GroupedData.apply).
 
+### Grouped Aggregate
+
+Grouped aggregate Pandas UDFs are similar to Spark aggregate functions. Grouped aggregate Pandas UDFs are used with groupBy and
+window operations. It defines an aggregation from one or more `pandas.Series`
+to a scalar value, where the `pandas.Series` represents values for a column within the same group or window.
+
+Note that this type of UDF doesn't not support partial aggregation and all data for a group or window will be loaded into memory. Also,
```
Comment (HyukjinKwon) [inline docs/sql-programming-guide.md]: Seems a typo `doesn't not` (BTW, I usually avoid abbreviation in documentation though).

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1804,6 +1804,25 @@ The following example shows how to use `groupby().apply()` to subtract the mean
...
+only unbounded window are supported with Grouped aggregate Pandas UDfs currently.
```
Comment (HyukjinKwon) [inline docs/sql-programming-guide.md]: `UDfs` -> `UDFs`

File: docs/sql-programming-guide.md [hunk 3]
```diff
@@ -1804,6 +1804,25 @@ The following example shows how to use `groupby().apply()` to subtract the mean
 For detailed usage, please see [`pyspark.sql.functions.pandas_udf`](api/python/pyspark.sql.html#pyspark.sql.functions.pandas_udf) and
 [`pyspark.sql.GroupedData.apply`](api/python/pyspark.sql.html#pyspark.sql.GroupedData.apply).
 
+### Grouped Aggregate
+
+Grouped aggregate Pandas UDFs are similar to Spark aggregate functions. Grouped aggregate Pandas UDFs are used with groupBy and
```
Comment (HyukjinKwon) [inline docs/sql-programming-guide.md]: `groupBy` -> `` `groupBy().agg()` ``

`window operations` -> can we link normal window Python API doc here?

**Comment by HyukjinKwon** [comment]
Seems fine otherwise

**Comment by icexelloss** [comment]
Thanks @HyukjinKwon ! I addressed the comments.

File: examples/src/main/python/sql/arrow.py [hunk 4]
```diff
@@ -113,6 +113,42 @@ def substract_mean(pdf):
...
+    # +---+----+------+
+    # | id|   v|mean_v|
+    # +---+----+------+
+    # |  1| 1.0|   1.5|
+    # |  1| 2.0|   1.5|
+    # |  2| 3.0|   6.0|
+    # |  2| 5.0|   6.0|
+    # |  2|10.0|   6.0|
+    # +---+----+------+
+    # $example off:grouped_map_pandas_udf$
```
Comment (HyukjinKwon) [inline examples/src/main/python/sql/arrow.py]: Seems a typo. It looks it should be `grouped_agg_pandas_udf`.  @icexelloss, `cd docs && SKIP_API=1 jekyll build` will build the doc and should better manually be tested.

Comment (icexelloss) [reply examples/src/main/python/sql/arrow.py]: Ah good catch, my bad. Let me try building the doc.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by icexelloss** [comment]
Thanks! @HyukjinKwon

**Pull request closed without merging**